test/unit/transaction-runner-test.coffee

```diff
@@ -538,3 +530,486 @@ describe 'TransactionRunner', ()->
+      emitter: new EventEmitter()
+      options:
+        'dry-run': false
+        method: []
+        header: []
+        reporter:  []
+        only: []
+        hookfiles: './**/*_hooks.*'
+
+    transaction = {}
+
+    beforeEach () ->
+      transaction =
+        name: 'Group Machine > Machine > Delete Message > Bogus example name'
+        id: 'POST /machines'
+        host: 'localhost'
+        port: '3000'
+        request:
+          body: '{\n  "type": "bulldozer",\n  "name": "willy"}\n'
+          headers:
+            'Content-Type': 'application/json'
+            'User-Agent': 'Dredd/0.2.1 (Darwin 13.0.0; x64)'
+            'Content-Length': 44
+          uri: '/machines'
+          method: 'POST'
+        expected:
+          headers: 'content-type': 'application/json'
+          body: '{\n  "type": "bulldozer",\n  "name": "willy",\n  "id": "5229c6e8e4b0bd7dbb07e29c"\n}\n'
+          statusCode: '202'
+        origin:
+          resourceGroupName: 'Group Machine'
+          resourceName: 'Machine'
+          actionName: 'Delete Message'
+          exampleName: 'Bogus example name'
+        fullPath: '/machines'
+
+      server = nock('http://localhost:3000').
+        post('/machines', {"type":"bulldozer","name":"willy"}).
+        reply transaction['expected']['statusCode'],
+          transaction['expected']['body'],
+          {'Content-Type': 'application/json'}
+      runner = new Runner(configuration)
+
+    afterEach () ->
+      nock.cleanAll()
+
+    describe 'with hooks', () ->
+      beforeEach () ->
+        sinon.spy loggerStub, 'info'
+        hooksStub.beforeHooks =
+          'Group Machine > Machine > Delete Message > Bogus example name' : [
+            (transaction) ->
+              loggerStub.info "before"
+          ]
+        hooksStub.afterHooks =
+          'Group Machine > Machine > Delete Message > Bogus example name' : [
+            (transaction, done) ->
+              loggerStub.info "after"
+              done()
+          ]
+
+      afterEach () ->
+        loggerStub.info.restore()
+
+      it 'should run the hooks', (done) ->
+        runner.executeAllTransactions [transaction], hooksStub, () ->
+          assert.ok loggerStub.info.calledWith "before"
+          assert.ok loggerStub.info.calledWith "after"
+          done()
+
+    describe 'with multiple hooks for the same transaction', () ->
+      beforeEach () ->
+        sinon.spy loggerStub, 'info'
+        hooksStub.beforeHooks =
+          'Group Machine > Machine > Delete Message > Bogus example name' : [
+            (transaction) ->
+              loggerStub.info "first",
+            (transaction, cb) ->
+              loggerStub.info "second"
+              cb()
+          ]
+
+      afterEach () ->
+        loggerStub.info.restore()
+
+      it 'should run all hooks', (done) ->
+        runner.executeAllTransactions [transaction], hooksStub, () ->
+          assert.ok loggerStub.info.calledWith "first"
```

Comment:
you are not checking order of calls.
I'd vote to use `loggerStub.info.firstCall.calledWith 'first'`or similar.